library/python/src/maps.cc

```diff
@@ -252,6 +252,68 @@ void add_map_bindings(nb::module_& m) {
                &HashedChunkedWaveletOctree::getCellValue, nb::const_),
            "node_index"_a,
            "Query the value of the map at a given octree node index.")
+      .def(
+          "get_cell_values",
+          [](const HashedChunkedWaveletOctree& self,
+             const nb::ndarray<IndexElement, nb::shape<-1, 3>, nb::device::cpu>&
+                 indices) {
+            // Create a query accelerator
+            QueryAccelerator<HashedChunkedWaveletOctree> query_accelerator{
+                self};
+            // Create nb::ndarray view for efficient access to the query indices
+            const auto index_view = indices.view();
+            const auto num_queries = index_view.shape(0);
+            // Create the raw results array and wrap it in a Python capsule that
+            // deallocates it when all references to it expire
+            auto* results = new float[num_queries];
+            nb::capsule owner(results, [](void* p) noexcept {
+              delete[] reinterpret_cast<float*>(p);
+            });
+            // Compute the interpolated values
+            for (size_t query_idx = 0; query_idx < num_queries; ++query_idx) {
+              results[query_idx] = query_accelerator.getCellValue(
+                  {index_view(query_idx, 0), index_view(query_idx, 1),
+                   index_view(query_idx, 2)});
+            }
+            // Return results as numpy array
+            return nb::ndarray<nb::numpy, float>{
+                results, {num_queries, 1u}, owner};
+          },
+          "index_list"_a,
+          "Query the map at the given indices, provided as a matrix with one "
+          "(x, y, z) index per row.")
+      .def(
+          "get_cell_values",
+          [](const HashedChunkedWaveletOctree& self,
+             const nb::ndarray<IndexElement, nb::shape<-1, 4>, nb::device::cpu>&
+                 indices) {
+            // Create a query accelerator
+            QueryAccelerator<HashedChunkedWaveletOctree> query_accelerator{
+                self};
+            // Create nb::ndarray view for efficient access to the query indices
+            auto index_view = indices.view();
+            const auto num_queries = index_view.shape(0);
+            // Create the raw results array and wrap it in a Python capsule that
+            // deallocates it when all references to it expire
+            auto* results = new float[num_queries];
```

Comment:
I assume that's nanobind that wants to have raw memory fun rather than raii things?

Reply:
Yes. I was also a bit surprised. Their docs seem to suggest this approach, and I couldn't find an RAII-based alternative.